Extinction of simple issues

eerussianguy opened the pull request:
Seatbelts, everyone.

Crop Changes:
- Wild Crops now spawn more mature based on the season, becoming mature around Late Summer
- Closes #1358 Crops now have different temperature and rainfall requirements to spawn and grow at different speeds
- These are enumerated on the spreadsheet, but to summarize:
- Maize, Rice, and Sugarcane prefer the hottest temperatures.
- Beets, Cabbage, Garlic, and Barley prefer colder temperatures. Garlic can survive up to -20 C.
- Some crops favor a wide range of temperature and rainfall, some are rather picky

Closes #1403 Added uses for Sticks:
- Burning a Bunch of Sticks produces 18 torches
- Armor stands can be crafted using a bunch of sticks and a smooth slab
- Crafting two bunches of sticks together vertically produces a Stick Bundle
- A Stick Bundle is for all intents and purposes a kind of log, except that it cannot be placed normally.
- Stick Bundles are a slightly below average fuel for Firepits
- Stick Bundles can be stored in Log Piles and therefore used to create Charcoal Pits

Closes #1435 changed how dyeing Alabaster and other blocks works
- The only blocks that are able to be barrel-dyed are the 'natural' states of each block, i.e. White Wool for Wool or Aggregate for Concrete Powder
- Blocks can be bleached into their natural form using the new liquid Lye, at a ratio of 125mB per block in a Barrel.
- Lye is made from 125 mB of Wood Ash (dropped by Firepits) in Spring Water

Closes #1091 Breaking glass has a 50% chance to drop Glass Shards, which can be melted back into Glass.

Firepit Changes:
- Changed model for the Firepit and the Ceramic Pot
- Firepit will make a sound when it's boiling or grilling
- Food on the grill will now show up in the world
- Closes #1115 Shift + Right-Click when the grill is empty to remove the attachment
- Added custom damage sources for attempting to remove a hot attachment from the Firepit
- Closes #1459 Ceramic Pots render water inside, which changes color to a reddish brown when finished cooking.
- Breaking a Firepit drops Wood Ash, used to make Lye.

Closes #1113 Added new particle effects:
- 3 new leaf particles, that drop when a leaf expires and float away.
- Bubbles from boiling soup
- Smoke from Grilling
- Sparks when welding an item on the anvil
- Firepit smoke has been split into 3 new particles, which look less like burning tires and more like smoke
- Decreased the overall speed of smoke rising
- Forges shoot out bits of lava and smoke randomly

Misc:
Closes #1535 Stone Anvils can be created if under any non-solid block (i.e. tool racks)
Closes #1519 Trapdoors are now crafted with 6 lumber to make 3 trapdoors
Closes #1494 Barrels will now show the recipe length in days if appropriate
Closes #1411 Anvils have a config for an "acceptable range" around the target where the smithing recipe will finish (default 0)
Closes #1399 Config to disable TFC hoes creating farmland
Closes #1376 Flux now shows on the anvil
Closes #1241 Glass is now a valid block for Charcoal Pits (can be disabled through config)
Closes #1090 Knapping a Crucible now produces an Unfired Crucible, which has to be cooked.
Closes #967 Breaking Ice with a Saw will drop the Ice block.
Closes #956 Jug filling has it's own sound event and custom sound.
Closes #931 Debris types (rocks, sticks, ores) can be disabled individually through config
Closes #792 Added Metal Trapdoors, which are smithed from Sheets.
Closes #588 Charcoal Pit explainer now appears in JEI when you click Charcoal

Fixes:
Closes #1541 Patchouli crash due to changed entity names
Closes #1540 Missing sound event localization
Closes #1532 Incorrect advancement localization
Closes #1530 bad model data for red steel shield
Closes #1531 Can't break lamp with hand (changed material to REDSTONE_LIGHT)
Closes #1522 Trapdoors rendered wrong
Closes #1510 Sluices allow placing torches on top

eerussianguy committed:
enhancement extinction

eerussianguy committed:
Merge remote-tracking branch 'upstream/1.12.x' into soviet_pp

eerussianguy committed:
fix patchouli

eerussianguy committed:
fix lamps and leaf particles

eerussianguy committed:
fixes

eerussianguy committed:
localization fixes for animal sounds

alcatrazEscapee commented on src/main/java/net/dries007/tfc/objects/items/ItemStickBundle.java:
```diff
@@ -0,0 +1,33 @@
+
+import javax.annotation.Nonnull;
+
+public class ItemStickBundle extends ItemTFC
```
This class is redundant. Just use `ItemMisc` and pass in the correct size / ore dictionary names. Mind the required format (e.g. `log_wood`, `stick_bundle`). See other examples for usage.

alcatrazEscapee commented on src/main/java/net/dries007/tfc/objects/blocks/devices/BlockFirePit.java:
```diff
@@ -273,6 +320,46 @@ else if (tile.getCookingPotStage() == TEFirePit.CookingPotStage.FINISHED)
+                    }
+                    else
+                    {
+                        tile.onRemoveAttachment();
+                        worldIn.setBlockState(pos, state.withProperty(ATTACHMENT, FirePitAttachment.NONE));
+                        ItemStack attachmentItem = ItemStack.EMPTY;
+                        if (attachment == FirePitAttachment.COOKING_POT)
+                            attachmentItem = ItemsTFC.FIRED_POT.getDefaultInstance();
+                        if (attachment == FirePitAttachment.GRILL)
+                            attachmentItem = ItemsTFC.WROUGHT_IRON_GRILL.getDefaultInstance();
+                        player.addItemStackToInventory(attachmentItem);
```
No, this needs to give the actual item instance back to the player that is held by the tile. That's *why it is held there*.

alcatrazEscapee replied:
Also, you CANNOT call `Item#getDefaultInstance` here as it is `@SideOnly(Side.CLIENT)`, which will crash and burn on dedicated servers.

alcatrazEscapee commented on src/main/java/net/dries007/tfc/objects/te/TEFirePit.java:
```diff
@@ -581,6 +583,11 @@ public void onConvertToGrill(EntityPlayer player, ItemStack stack)
         attachedItemStack = stack.splitStack(1);
     }
 
+    public void onRemoveAttachment()
+    {
+        attachedItemStack = ItemStack.EMPTY;
+    }
+
```
No, this should handle dropping and giving back to player, as the attachedItemStack needs to be given back to the player exactly.

eerussianguy committed:
Review changes:
- stick bundle no longer has it's own class
- firepits now remove attachment safely

eerussianguy committed:
changes conditions for stone anvil creation

eerussianguy committed:
FINALLY fixed the trapdoors

HeartyYF commented on src/main/resources/assets/tfc/lang/en_us.lang:
```diff
@@ -4764,8 +4819,14 @@ tfc.animal.boar.cry=Boar cries
+tfc.animal.wildebeest.death=Wildebeest dies
 tfc.animal.quail.say=Quail chirps
+tfc.animal.quail.hurt=Quail hurts
+tfc.animal.quail.death=Qual dies
```
`Quail`

HeartyYF commented on src/main/resources/assets/tfc/patchouli_books/book/en_us/entries/food/animal_husbandry.json:
```diff
@@ -23,13 +23,13 @@
 			"recipe": "tfc:nest_box",
 			"title": "Nest Box",
-			"text": "Chickens lay eggs daily if you give them a $(item)nest box$() to use. As long as a rooster is near the nest box when the egg is laid, the egg will become fertilized and will hatch eventually. Eggs can also be cooked for food."
+			"text": "Chickens, Grouses, and Quails lay eggs daily if you give them a $(item)nest box$() to use. As long as a rooster is near the nest box when the egg is laid, the egg will become fertilized and will hatch eventually. Eggs can also be cooked for food."
```
I suggest mentioning ducks

HeartyYF commented on src/main/resources/assets/tfc/patchouli_books/book/en_us/entries/getting_started/predators.json:
```diff
@@ -50,6 +50,48 @@
+    },
+    {
+      "type": "entity",
+      "name": "Panther",
```
Another panther. Maybe it should be something else, but seems only 5 predators were added.

HeartyYF commented on src/main/resources/assets/tfc/patchouli_books/book/en_us/entries/getting_started/predators.json:
```diff
@@ -50,6 +50,48 @@
+    },
+    {
+      "type": "entity",
+      "name": "Panther",
```
The third panther.

eerussianguy committed:
fixed patchouli and lang

DisasterMoo commented on src/main/resources/assets/tfc/patchouli_books/book/en_us/entries/food/animal_husbandry.json:
```diff
@@ -29,7 +29,7 @@
 			"recipe": "tfc:nest_box",
 			"title": "Nest Box",
-			"text": "Chickens, Grouses, and Quails lay eggs daily if you give them a $(item)nest box$() to use. As long as a rooster is near the nest box when the egg is laid, the egg will become fertilized and will hatch eventually. Eggs can also be cooked for food."
+			"text": "Chickens, Grouses, Ducks, and Quails lay eggs daily if you give them a $(item)nest box$() to use. As long as a rooster is near the nest box when the egg is laid, the egg will become fertilized and will hatch eventually. Eggs can also be cooked for food."
```
```suggestion
			"text": "Chickens, Grouses, Ducks, and Quails lay eggs daily if you give them a $(item)nest box$() to use. As long as a male from the same species mate with the female before she lay eggs, the egg will become fertilized and will hatch eventually. Eggs can also be cooked for food."
```

eerussianguy committed:
Update src/main/resources/assets/tfc/patchouli_books/book/en_us/entries/food/animal_husbandry.json

Co-authored-by: DisasterMoo <luan.santos.pinto@gmail.com>

DisasterMoo commented on src/main/java/net/dries007/tfc/objects/te/TEAnvilTFC.java:
```diff
@@ -240,7 +241,9 @@ public void addStep(@Nonnull EntityPlayer player, @Nullable ForgeStep step)
-                if (workingProgress == workingTarget && completedRecipe.matches(steps))
+                int workMin = workingTarget - ConfigTFC.General.DIFFICULTY.acceptableAnvilRange;
+                int workMax = workingTarget + ConfigTFC.General.DIFFICULTY.acceptableAnvilRange;
+                if ((workingProgress < workMax && workingProgress > workMin) && completedRecipe.matches(steps))
```
```suggestion
                if ((workingProgress <= workMax && workingProgress >= workMin) && completedRecipe.matches(steps))
```
otherwise, this will make it impossible to finish if both `workMin` and `workMax` are the same (default config = 0)

eerussianguy committed:
Update src/main/java/net/dries007/tfc/objects/te/TEAnvilTFC.java

Co-authored-by: DisasterMoo <luan.santos.pinto@gmail.com>
